crates/astria-conductor/src/data_availability/sync.rs

```diff
@@ -0,0 +1,191 @@
+use std::time::Duration;
+
+use celestia_client::{
+    celestia_types::{
+        nmt::Namespace,
+        Height,
+    },
+    jsonrpsee::http_client::HttpClient,
+    CelestiaClientExt as _,
+    CelestiaSequencerBlob,
+};
+use color_eyre::eyre::{
+    self,
+    Report,
+    WrapErr as _,
+};
+use futures::stream::FuturesOrdered;
+use tokio::select;
+use tracing::{
+    error,
+    info,
+    instrument,
+    warn,
+};
+use tryhard::RetryFutureConfig;
+
+use super::block_verifier::BlockVerifier;
+use crate::{
+    data_availability::{
+        send_sequencer_subsets,
+        verify_sequencer_blobs_and_assemble_rollups,
+    },
+    executor,
+};
+
+pub(crate) async fn find_da_block_with_sequencer_data(
+    start_height: Height,
+    celestia_client: HttpClient,
+    sequencer_namespace: Namespace,
+    da_block_range: u32,
+) -> Height {
+    let mut height = start_height;
+    let mut loop_count = 0;
+    loop {
+        assert!(
+            loop_count <= da_block_range,
+            "{}",
+            format!("initial sequencer block not found after searching {da_block_range} da blocks")
+        );
+        let sequencer_blobs = celestia_client
+            .get_sequencer_blobs(height, sequencer_namespace)
+            .await;
+
+        match sequencer_blobs {
+            Ok(_blobs) => {
+                info!(height = %height.value(), "found sequencer blob");
+                break height;
+            }
+            Err(error) => {
+                warn!(height = %height.value(), error = %error, "error returned when fetching sequencer data; skipping");
+                height = height.increment();
+                loop_count += 1;
+                continue;
+            }
+        }
+    }
+}
+
+/// Fetch sequencer data from DA with at given height and namespace with
+/// retries. The fetch will be retried 10 times with exponential backoff before
+/// failing.
+pub(crate) async fn get_new_sequencer_block_data_with_retry(
+    height: Height,
+    celestia_client: HttpClient,
+    sequencer_namespace: Namespace,
+) -> Result<Vec<CelestiaSequencerBlob>, Report> {
+    let retry_config = RetryFutureConfig::new(10)
+        .exponential_backoff(Duration::from_secs(5))
+        .max_delay(Duration::from_secs(60))
+        .on_retry(|attempt, next_delay: Option<Duration>, error: &Report| {
+            let wait_duration = next_delay
+                .map(humantime::format_duration)
+                .map(tracing::field::display);
+            let error: &(dyn std::error::Error + 'static) = error.as_ref();
+            let error = error.to_string();
```

Comment:
please don't do ths. log the error using its `tracing::field::Value` implementation